### PR TITLE
Rework

### PR DESCRIPTION
**Pull request by solnic**
### Summary

* `Add HTTP::Schema` and `HTTP::Attribute`
* Add custom `HTTP::MapperCompiler`
* Remove manual transformations in favor of auto-mapping feature from core
* Remove applying transformations in insert/update - this type of functionality is not supported by core and it's actually something we intentionally do not want to have. Input that doesn't match canonical data structures can be handled by changesets with their mappings
* Refactor `Dataset` so that it only uses its options
* Remove `Dataset#name` as it felt like additional complexity
* Remove `Dataset::ClassInterface` (we're in beta phase, no need to go through deprecation cycles)
* Add `Dataset#{get?,post?,put?,delete?}`
* Remove `HTTP::Relation#wrap` as this is now a core method and it does something else
* Simplify specs

### CODE REVIEW

**Comment by solnic** [comment]
@AMHOL @cflipse @flash-gordon @timriley ding dong, please check this out. I'd love to get it done and add a couple features (associations mostly and a built-in json adapter specialization) on top of this rework.

**Comment by cflipse** [comment]
Looks okay so far

File: lib/rom/http/relation.rb [hunk 1]
```diff
@@ -1,40 +1,34 @@
-require 'dry/core/cache'
 require 'rom/initializer'
-require 'rom/http/transformer'
+
+require 'rom/http/attribute'
+require 'rom/http/schema/dsl'
+require 'rom/http/schema/dsl'
```
Comment (flash-gordon) [inline lib/rom/http/relation.rb]: repetitive `require`

File: lib/rom/http/relation.rb [hunk 2]
```diff
@@ -1,40 +1,34 @@
...
 
       def primary_key
-        attribute = schema.find(&:primary_key?)
-
-        if attribute
-          attribute.alias || attribute.name
-        else
-          :id
-        end
+        schema.primary_key[0].name
```
Comment (flash-gordon) [inline lib/rom/http/relation.rb]: I think we can have a missing method here, mb it's a good idea to raise a meaningful exception

**Comment by AMHOL** [comment]
@solnic can this branch be deleted?

**Pull request merged**